pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedProducerImpl.java

```diff
@@ -438,4 +438,13 @@ public Timeout getPartitionsAutoUpdateTimeout() {
         return partitionsAutoUpdateTimeout;
     }
 
+    /**
+     * Return the number of partitions per topic.
+     *
+     * @return the number of partitions per topic.
+     */
+    public int getNumOfPartitions() {
```

Comment:
Do we need to expose this method to the `Producer` interface? I think we should not let the user call this method in the implementation class directly.

Reply:
If so, how to handle the non-partitioned topics `ProducerImpl`? Does it mean the non-partitioned topic's partition number is 1 ? @RobertIndie @codelipenghui

Reply:
I think we can return `0` in that case. Because `1` denotes a partitioned topic with 1 partition but not the non-partitioned topic. But I would like to hear others' opinions.

Reply:
+1, for non-partitioned topics, the partition number should be 0. This behavior is consistent with what `Topics#getPartitionedTopicMetadata` returns.

Reply:
@RobertIndie @BewareMyPower Thank your good suggestion. I updated my pull request. Please help to review it.